### DISCUSSION TITLE
ignore ways that are already covered by relation these ways belong to

### DISCUSSION THREAD

**Discussion by 62mkv** [EqualStreetNames/equalstreetnames#757] [Q&A]
I am processing city of Narva and I have quite few instances like this: 

https://www.openstreetmap.org/way/39909952#map=16/59.37132/28.15288

which is part of https://www.openstreetmap.org/relation/11258203#map=15/59.37423/28.16851

as you can see, someone really put effort into assigning wikidata tags directly on ways: https://www.openstreetmap.org/changeset/120684481#map=15/59.37733/28.16836

I don't feel right about completely removing this person's work.. 

but what we have now is that once wikidata items for relations are properly edited with "named after" statement added; the `gender.csv` will have duplicate entries, both for relation and way. which obviously does not feel right

I am not sure what kind of an impact it will have when deployed but I assume at least the stats will be skewed; so I presume there should be a way to avoid this kind of things (by giving preference to relations over ways, in cases when way and relation are ultimately named after same person, I presume)

**Comment by jbelien** [comment]
Hello @62mkv ,

That's a good point! But that's already part of the process. 😃 
When the process runs, it makes a group by streetname. So it doesn't matter if it's tagged on the relation or on the way, if the **streetname** is the same, it will be merge together during the process (and receive the type `relation+way`, see https://github.com/EqualStreetNames/module-process/blob/master/Command/StatisticsCommand.php#L347).

The "issue" here is that the [way](https://www.openstreetmap.org/way/39909952) doesn't have the same name ("Paul Kerese") as the [relation](https://www.openstreetmap.org/relation/11258203) ("Paul Kerese tänav").
The whole project is based on the streetname and so in this case, those 2 "objects" won't be merged together.

For reference: https://github.com/EqualStreetNames/module-process/blob/master/Command/StatisticsCommand.php#L279-L353

**Reply by 62mkv** [reply]
Thanks for your response!! 

Pardon my ignorance but we do know which ways are part of which relation, so if we have enough data on relation, should we not just ignore ways that are part of that (and point to the same person anyway)?

**Reply by 62mkv** [reply]
I believe there is a similar test that checks if a way and relation it is part of have same or different etymology tag.. probably could extend this to just mark the way as excluded if it has same etymology as relation (or same etymology as relation wikidata item is "named after")

**Reply by jbelien** [reply]
I'll check if I could add some extra checked like that. 👍
